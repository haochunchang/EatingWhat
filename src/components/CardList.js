import React, { Fragment } from "react";
import Card from "./Card";

const CardList = ({ foods }) => {
  const cardComponent = foods.map((food, i) => {
    return (
      <Card
        key={i}
        id={foods[i].id}
        name={foods[i].name}
        category={foods[i].category}
      />
    );
  });
  return <Fragment>{cardComponent}</Fragment>;
};

export default CardList;
