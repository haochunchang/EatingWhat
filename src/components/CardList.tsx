import React, { Fragment } from "react";
import Card from "./Card";
import { Food } from "./MainPage";

const CardList = ({ foods }: { foods: Array<Food> }) => {
  const cardComponent = foods.map((food, i) => {
    return <Card key={i} url={foods[i].url} name={foods[i].name} />;
  });
  return <Fragment>{cardComponent}</Fragment>;
};

export default CardList;
