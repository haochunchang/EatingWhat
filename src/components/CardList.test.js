import React from "react";
import { shallow } from "enzyme";
import CardList from "./CardList";

describe("CardList", () => {
  it("matches CardList component snapshot", () => {
    const mockFoods = [
      {
        id: 1,
        name: "noodles",
        url: "https://whereismyfood.com",
        category: "main food",
      },
    ];
    expect(shallow(<CardList foods={mockFoods} />)).toMatchSnapshot();
  });
});
