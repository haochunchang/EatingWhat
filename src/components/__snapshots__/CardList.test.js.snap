// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList matches CardList component snapshot 1`] = `ShallowWrapper {}`;
